ChessStatePattern/src/states/State.java

```diff
@@ -1,16 +1,36 @@
 package states;
 
 import main.*;
+import states.commons.Move;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public abstract class State {
 	
 	ChessGame game;
-	
+	String activePlayer;
+	List<Move> moveHistory;
```

Comment:
In the state pattern, each state’s primary role is to define behavior, not to store or manage cross-state data. ChessGame as the main controller manages game-wide data, while state classes focus on defining state-specific behavior.

In this case, what do you think about moving this to ChessGame?